File: app/src/test/java/com/farmerbb/taskbar/util/RunnableHooker.java
package com.farmerbb.taskbar.util;

/**
 * {@link RunnableHooker} for test.
 * <p>
 * It provides method to check whether the {@link Runnable#run} method has been invoked.
 * <p>
 * Also, the powermock can't process the code likes:
 *
 * <pre>
 *     Runnable runnable = new Runnable() {
 *         @Override
 *         public void run() {}
 *     }
 * </pre>
 * <p>
 * or
 *
 * <pre>
 *     Runnable runnable = () -> {}
 * </pre>
 * <p>
 * So we create a implementation to avoid powermock exception.
 */
public class RunnableHooker implements Runnable {
    private boolean hasRun = false;

    @Override
    public void run() {
        hasRun = true;
    }

    public boolean hasRun() {
        return hasRun;
    }
}

File: app/src/test/java/com/farmerbb/taskbar/util/RunnableHooker.kt
package com.farmerbb.taskbar.util

/**
 * [RunnableHooker] for test.
 *
 *
 * It provides method to check whether the [Runnable.run] method has been invoked.
 *
 *
 * Also, the powermock can't process the code likes:
 *
 * <pre>
 * Runnable runnable = new Runnable() {
 * @Override
 * public void run() {}
 * }
 * </pre>
 *
 *
 * or
 *
 * <pre>
 * Runnable runnable = () -> {}
 * </pre>
 *
 *
 * So we create a implementation to avoid powermock exception.
 */
class RunnableHooker : Runnable {
    private var hasRun = false
    override fun run() {
        hasRun = true
    }

    fun hasRun(): Boolean {
        return hasRun
    }
}
